src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java

```diff
@@ -111,24 +111,22 @@ public class GencodeFuncotationFactory extends DataSourceFuncotationFactory {
     /**
      * List of valid Appris Ranks used for sorting funcotations to get the "best" one.z
      */
-    private static final HashSet<GencodeGtfGeneFeature.FeatureTag> apprisRanks = new HashSet<>(
-            Arrays.asList(
-                APPRIS_PRINCIPAL,
-                APPRIS_PRINCIPAL_1,
-                APPRIS_PRINCIPAL_2,
-                APPRIS_PRINCIPAL_3,
-                APPRIS_PRINCIPAL_4,
-                APPRIS_PRINCIPAL_5,
-                APPRIS_ALTERNATIVE_1,
-                APPRIS_ALTERNATIVE_2,
-                APPRIS_CANDIDATE_HIGHEST_SCORE,
-                APPRIS_CANDIDATE_LONGEST_CCDS,
-                APPRIS_CANDIDATE_CCDS,
-                APPRIS_CANDIDATE_LONGEST_SEQ,
-                APPRIS_CANDIDATE_LONGEST,
-                APPRIS_CANDIDATE
-            )
-    );
+    private static final LinkedHashMap<String, GencodeGTFFieldConstants.FeatureTag> apprisRanks = new LinkedHashMap<>() {{
```

Comment:
@jonn-smith i'm halfway tempted to create an entirely new enum for appris rank since its one of the only ones that actually has to get parsed and handled seperately and it would free us from needing the FeatureTag list